dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -531,37 +553,55 @@ public ListenableFuture<List<LoadFileFailure>> load(
       // These two need UFS api support and cannot be achieved in a generic UFS interface.
       // We may be able to solve this by providing specific implementations for certain UFSes
       // in the future.
-      if (loadData && status.isFile() && (status.asUfsFileStatus().getContentLength() > 0)) {
-        try {
-          ListenableFuture<Void> loadFuture = Futures.submit(() -> {
-            try {
-              if (options.hasUser()) {
-                AuthenticatedClientUser.set(options.getUser());
-              }
-              loadData(status.getUfsFullPath().toString(), 0,
-                  status.asUfsFileStatus().getContentLength());
-            } catch (Throwable e) {
-              LOG.error("Loading {} failed", status, e);
-              boolean permissionCheckSucceeded = !(e instanceof AccessControlException);
-              AlluxioRuntimeException t = AlluxioRuntimeException.from(e);
-              errors.add(LoadFileFailure.newBuilder().setUfsStatus(status.toProto())
-                  .setCode(t.getStatus().getCode().value())
-                  .setRetryable(t.isRetryable() && permissionCheckSucceeded)
-                  .setMessage(t.getMessage()).build());
+      if (!loadData || !status.isFile()) {
+        continue;
+      }
+      long fileLength = status.asUfsFileStatus().getContentLength();
+      boolean countAsSkipped =
+          skipIfExists && isAllPageCached(status.asUfsFileStatus(), originalFs);
+      if (countAsSkipped) {
+        skippedFiles.incrementAndGet();
+        skippedFileLength.addAndGet(status.asUfsFileStatus().getContentLength());
+        continue;
+      }
+      if (fileLength == 0) {
+        continue;
+      }
+      try {
+        ListenableFuture<Void> loadFuture = Futures.submit(() -> {
+          try {
+            if (options.hasUser()) {
+              AuthenticatedClientUser.set(options.getUser());
             }
-          }, GrpcExecutors.READER_EXECUTOR);
-          futures.add(loadFuture);
-        } catch (RejectedExecutionException ex) {
-          LOG.warn("BlockDataReaderExecutor overloaded.");
-          AlluxioRuntimeException t = AlluxioRuntimeException.from(ex);
-          errors.add(LoadFileFailure.newBuilder().setUfsStatus(status.toProto())
-              .setCode(t.getStatus().getCode().value())
-              .setRetryable(true)
-              .setMessage(t.getMessage()).build());
-        }
+            loadData(status.getUfsFullPath().toString(), 0,
+                status.asUfsFileStatus().getContentLength());
+          } catch (Throwable e) {
+            LOG.error("[DistributedLoad] Loading {} failed", status, e);
+            AlluxioRuntimeException t = AlluxioRuntimeException.from(e);
+            errors.add(LoadFileFailure.newBuilder().setUfsStatus(status.toProto())
+                .setCode(t.getStatus().getCode().value())
+                .setRetryable(true)
```

Comment:
why always set retryable as true? Then what's the meaning of retryable...

Reply:
because this flag is not reliable.... we are not able to tell if an exception is really retry-able by just looking at t.isRetryable().
This one caused a couple of issues when i used this tool previously. Either ignoring retryable errors or doing some endless retries...
If you want to improve this. I think you can default the retryable to true and only add a couple of non-retryable "exceptions"